db/migrate/20230423202740_hide_email_by_default.rb

```diff
@@ -0,0 +1,17 @@
+class HideEmailByDefault < ActiveRecord::Migration[7.0]
```

Comment:
I would prefer we do this in multiple steps:

1) add new `public_email` column
2) dual write both columns
3) backfill new column
4) read only from new column
5) delete old column

otherwise, there is the potential that any mistake will leave us with irrecoverable data changes

Reply:
If I understand it well, whole up/down blocks are wrapped in transaction and PostgreSQL schema is mostly transactional. That means whole block is executed or nothing leaving no space for unpredictable state. :thinking: But I can change to suggested format if needed.

Reply:
I don't feel too strongly but I also prefer to split these kinds of data changes into multiple steps as @segiddins described.

Reply:
@segiddins I'm about to merge this. If I understand it well (by recent experience), we will need to do it exactly as you mentioned (using `ignored_columns` trick) to make smooth deployment. Can you confirm please? I'll update the PR.

Reply:
please rebase & see if strong_migrations errors

Reply:
It does not by default, since this migration has old timestamp. But setting `StrongMigrations.start_after = 0` and trying to run this migration errors thanks to `strong_migrations`.